spec/models/shopify/client_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+RSpec.describe Shopify::Client, type: :model do
+  describe 'connect to shop', vcr: { cassette_name: 'shopify/client' } do
+    let(:client) { described_class.new(credentials) }
+
+    context 'with valid credentials' do
+      let(:credentials) { { api_key: 'baz',
+                            password: 'foo',
+                            shop_name: 'test_shop' } }
+      let(:result) { { "count" => 1 } }
+
+      it 'returns number of products' do
+        expect(client.total_products).to eq(result)
+      end
+    end
+
+    context 'with invalid credentials' do
+      let(:credentials) { { api_key: 'foo', password: 'baz', shop_name: 'fake-shop' } }
+
+      it 'raise forbidden access error' do
+        expect { client.total_products }.to raise_error ActiveResource::ClientError
```

Comment:
Can you check for the message here?I assume  `ClientError` could happen for many different reasons like: invalid credentials, unsupported API call, server timeout, server error,...